### PR TITLE
remove raydp dependency

### PR DESCRIPTION
**Pull request by anyscalesam**
## Why are these changes needed?

this removes the raydp dependency from our ray-ml requirements.txt list to mitigate bringing in an older version of pyspark which in turn brings other older versions of dependent libraries.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
Could you update the pr description of why we remove it.

**Comment by aslonnie** [comment]
@anyscalesam  you would need to checkin the recompiled requirement files.

**Comment by anyscalesam** [comment]
@aslonnie i don't see any *recompiled.txt with instance of "raydp" can you point me to the one?

File: python/requirements_compiled.txt [hunk 1]
```diff
@@ -25,23 +25,23 @@ absl-py==1.4.0
     #   tensorflow-probability
     #   tf-slim
 accelerate==0.28.0
-    # via -r /home/ubuntu/ray/ci/../python/requirements/ml/core-requirements.txt
+    # via -r /ray/ci/../python/requirements/ml/core-requirements.txt
```
Comment (aslonnie) [inline python/requirements_compiled.txt:28]: @can-anyscale changed to `/ray/ci` stuff.

File: python/requirements_compiled.txt [hunk 2]
```diff
@@ -1851,22 +1842,18 @@ qpd==0.4.4
     # via fugue
 querystring-parser==1.2.4
     # via mlflow
-    # via
-    #   raydp
-    #   tune-sklearn
-raydp==1.7.0b20231020.dev0
-    # via -r /home/ubuntu/ray/ci/../python/requirements/ml/data-requirements.txt
```
Comment (aslonnie) [inline python/requirements_compiled.txt:1858]: raydp is gone

not sure if data wants to add this into test dependencies or not.

**Comment by aslonnie** [comment]
I think we still have quite some raydp tests.. probably need to move it into test deps

**Review by c21 — APPROVED** [review]

File: python/requirements_compiled.txt (hunk 2 above)
Comment (c21) [reply python/requirements_compiled.txt:1858]: yes still want it for test dependency.

**Comment by aslonnie** [comment]
prefers https://buildkite.com/ray-project/premerge/builds/24913

**Pull request closed without merging**